credentials/credentials.go

```diff
@@ -120,6 +120,10 @@ type AuthInfo interface {
 	AuthType() string
 }
 
+type Validate interface {
```

Comment:
are we making sure this is an optional interface? If yes, how?

Reply:
This is a optional interface. There is no way to explicitly specify the interface as optional. Any credential that wants to allow authority override will have to implement the method, if the credentials do not want to allow authority override, it will not implement this interface. If we try to use the credentials not allowing the override and try to override, it will fail.

Reply:
yeah but you are making it optional by putting it under if condition of authority option. That's what i was asking but in actual PR it would be good to put in the comment.

Reply:
Okay